core/lib/dal/src/models/storage_block.rs

```diff
@@ -55,38 +54,41 @@ pub(crate) struct StorageL1BatchHeader {
     pub pubdata_input: Option<Vec<u8>>,
 }
 
-impl From<StorageL1BatchHeader> for L1BatchHeader {
-    fn from(l1_batch: StorageL1BatchHeader) -> Self {
-        let priority_ops_onchain_data: Vec<_> = l1_batch
+pub trait IntoL1BatchHeader {
```

Comment:
Bikeshedding: IMO, this trait is redundant; you can define `into_l1_batch_header_with_logs()` methods as inherent in `StorageL1BatchHeader` / `StorageL1Batch`. It would be slightly simpler.

Reply:
won't it be confucing?

Reply:
Why would it be confusing? Different structs can define methods with the same name especially if the have the same semantics.